[Kaptain] cert-manager workaround (#4545)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.0.0/release-notes/2.0.0/index.md

- Kubeflow Pipelines 1.8.1
- Training Operator 1.4.0
- Tensorflow 2.8.0
- PyTorch 1.11.0
- CUDA 11.4
- MXNet 1.9
- Horovod 0.24.2

> ## Known issues

Reviewer: should this be here or in the release notes for 2.1.0?
(I assume that it should be here, but just double confirming)

Author: We are trying to work on a solution for Kap 2.1, so we are not documenting the workaround for 2.1 (yet).

Reviewer: yeah that makes sense. happy with this remaining in docs for 2.0.0 👌

### cert-manager workaround for Kaptain

> Due to an oversight, some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications, which also affects Kaptain. While the effects can vary, the most common failure is the inability to launch Kaptain notebooks in Jupyter.

Reviewer 2: ```suggestion
Some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications, which also affects Kaptain. While the effects can vary, the most common failure is the inability to launch Kaptain notebooks in Jupyter.
```

#### Regenerate the secrets in DKP

A permanent fix for the issue requires upgrading to Kommander 2.2.1 or higher. If you are running other versions of DKP, refer to the cert-manager expiration workaround documentation for DKP [2.1.0](../../../../kommander/2.1/release-notes/2.1.0#cert-manager-expiration-workaround), [2.1.1](../../../../kommander/2.1/release-notes/2.1.1#cert-manager-expiration-workaround) or [2.1.2](../../../../kommander/2.1/release-notes/2.1.2#cert-manager-expiration-workaround) to run a docker container that extends the validity of the Cluster CA to 10 years and fixes the certificate reload issue.

> Once this is done, you can fix the issue on Kaptain’s side.

Reviewer 2: ```suggestion
Once this is done, the issue can be fixed on Kaptain’s side.
```

#### Regain access to Kaptain

This gives you back the capability of launching notebooks in Jupyter:

> 1.  Kaptain has one certificate that you have to delete to force a refresh, and one that you can update manually for Istio:

Reviewer 2: ```suggestion
1.  Kaptain has one certificate that most be deleted to force a refresh and one that can be updated manually for Istio:
```
